src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -745,21 +750,19 @@ export class Kernel implements IKernel {
 
             result.push(...changeDirScripts);
 
-            // Set the ipynb file
-            const file = this.resourceUri?.fsPath;
-            if (file) {
-                result.push(`__vsc_ipynb_file__ = "${file.replace(/\\/g, '\\\\')}"`);
-            }
             result.push(CodeSnippets.disableJedi);
 
             // For Python notebook initialize matplotlib
             result.push(...this.getMatplotLibInitializeCode());
 
             result.push(...debugCellScripts);
-        }
 
-        // Run any startup commands that we have specified
-        result.push(...this.getStartupCommands());
+            // Set the ipynb file
+            const file = this.resourceUri?.fsPath;
+            if (file) {
+                result.push(`__vsc_ipynb_file__ = "${file.replace(/\\/g, '\\\\')}"`);
```

Comment:
No try except here, if this fails, we'll end up logging this, better than adding a try except and we'll never ever see any failures.